src/widgets/index/index.ts

```diff
@@ -402,11 +410,19 @@ const index = (props: IndexProps): Index => {
       // widgets. When the subscription happens before the `init` step, the (static)
       // configuration of the widget is pushed in the URL. That's what we want to avoid.
       // https://github.com/algolia/instantsearch.js/pull/994/commits/4a672ae3fd78809e213de0368549ef12e9dc9454
-      helper.on('change', ({ state }) => {
-        localUiState = getLocalWidgetsState(localWidgets, {
-          searchParameters: state,
-          helper: helper!,
-        });
+      helper.on('change', ({ state, isPageReset }) => {
+        if (isPageReset) {
+          localUiState.page = undefined;
+        }
+
+        localUiState = getLocalWidgetsState(
+          localWidgets,
+          {
+            searchParameters: state,
+            helper: helper!,
+          },
+          localUiState
```

Comment:
If I understand https://github.com/algolia/instantsearch.js/pull/4359#issuecomment-623964835 well, you want to pass `localUiState` only at first render, and `{}` next times. Is that right? If not, can you explain when we need to use the current UI state and when we don't want to use it?